src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/SlimefunItem.java

```diff
@@ -344,6 +345,7 @@ public boolean isItem(ItemStack item) {
 		if (this instanceof ChargableItem && SlimefunManager.isItemSimiliar(item, this.item, false)) return true;
 		else if (this instanceof DamagableChargableItem && SlimefunManager.isItemSimiliar(item, this.item, false)) return true;
 		else if (this instanceof ChargedItem && SlimefunManager.isItemSimiliar(item, this.item, false)) return true;
+		else if ((SlimefunManager.isItemSimiliar(item, SlimefunItems.BROKEN_SPAWNER, false)) || (SlimefunManager.isItemSimiliar(item, SlimefunItems.REPAIRED_SPAWNER, false))) return true;
```

Comment:
This is not a static method.
Check if this.getID() matches BROKEN_SPAWNER or REINFORCED_SPAWNER for each of the two cases.

Reply:
While "this.getID()" is sometimes a BROKEN or REINFORCED spawner, it never is when interfacing with the anvil, and there doesn't appear to be a getID() for the item being checked.  The Very next line which is " return SlimefunManager.isItemSimiliar(item, this.item, true); " suggests that this was the proper thing to put there since this is part of the current release and supposedly works for everything else.  Also, this works in this form, if I start checking "this", it stops trying to match any item that the player is using/holding/whatever... and instead checks whatever item was last referenced by whichever function was last used... providing virtually random information that has nothing to do with the item I'm putting in the anvil (which is item).  I may still be doing something wrong here, but "this is not a static method" doesn't make sense to me since the very next line does nearly the identical thing, and that was just fine for the release.

Reply:
public boolean isItem(...

It isn't static. I don't really get what you are trying to explain, I think you are thinking a bit too complicated.

Reply:
I can understand if you want to undo this change as well, so I can at least merge the other changes for now.
Pull Requests don't have to be packed full of tweaks.

Reply:
"what" isn't static, what exactly are you talking about here, I didn't redefine anything here or change the method in any way?  And why does this work if it doesn't "work"?

Reply:
The method isn't static. Yes, you didn't redefine anything.

But the method belongs to an object, an instance of SlimefunItem.
And you are performing a check that does not include the current instance.

e.g. (again just pseudo)

SlimefunItem item = "Magic Eye Of Ender";
item.isItem(Broken Spawner) => true

SlimefunItem item = "Grandmas Walking Stick";
item.isItem(Broken Spawner) => true

It should only return true, if "item" was actually a Broken Spawner, the method you are working is a comparison and you just return true if it is a spawner, without checking "this".

Reply:
"this" cannot be accessed in a static context... I can't very well change the whole method to static, that would completely break it.  I really don't get what you're wanting changed here.

Reply:
> "this" cannot be accessed in a static context...

Again, the method is NOT static, so you not only can but must use "this"...
I want you to use "this", re-read my example above.

Reply:
You basically need to replace this line with this:

else if (this.getID().equals("BROKEN_SPAWNER")) return SlimefunManager.isItemSimiliar(item, SlimefunItems.BROKEN_SPAWNER, false);
else if (this.getID().equals("REINFORCED_SPAWNER")) return SlimefunManager.isItemSimiliar(item, SlimefunItems.REINFORCED_SPAWNER, false);

That's what it has to be, you need to use this in order to refer to the instance because this is object-oriented.
You can just use that snippet as it is, feel free to ask questions if you still don't understand why it needs to be this way.

Reply:
The result of this change is that it ceases to function and goes back to being broken again (broken spawners can be renamed on the anvil, bypassing the altar and all ingredients to become a placeable pig spawner).  I don't get why so caught up on this point.  In the two cases where this needs to return TRUE and no other thing... an item has been passed to this method which 
 -- >SlimefunManager.isItemSimiliar(item, SlimefunItems.REPAIRED_SPAWNER, false)  <-- 
The result of this statement is also TRUE.  Why are you having me check to make sure that "this...." anything??? It has nothing to do with what needs to be done here.  I'm not comparing to "this" like other methods may be doing and still need to do... in THIS CASE I need it to return "true", regardless of what "this" is, the item has been established to be a reinforced or broken spawner, isItem necessarily needs to return TRUE for those items.

Reply:
Also I've tried doing a lot of echoing and things, and from the best of my knowledge at the time of the check when the spawner is placed into the anvil and attempted to be renamed... this.getID() is either MOB_SPAWNER or ANVIL.

Reply:
Show me that code.

Reply:
...
else if ((SlimefunManager.isItemSimiliar(item, SlimefunItems.BROKEN_SPAWNER, false)) || (SlimefunManager.isItemSimiliar(item, SlimefunItems.REPAIRED_SPAWNER, false))) return true;

Right here... if item is either of these, this is true.  No use of this becasue that's not what's being compared.

as taken from ItemListener.java....
public void onAnvil(InventoryClickEvent e) { ... 
if .. !SlimefunItem.isDisabled ... 

in this very same file SlimefunItem.java:
	public static boolean isDisabled(ItemStack item) {
	    for (SlimefunItem i: all) {
			if (i.isItem(item)) {
				return i.isDisabled();
			}
		}
	    return false;
	}


return i.isItem(item) --> this needs to be true for REINFORCED and BROKEN spawners, this is all that needs to happen here, this is what my one added line of code does.

Reply:
> return i.isItem(item) --> this needs to be true for REINFORCED and BROKEN spawners
Yes but it has to be true ONLY if they actually are Spawners...
Returning true no matter what doesn't fix anything, it just breaks everything.

And as you pointed out, the for loop does "i.isItem(item)"
In other words this compares "i" to "item".
But you only check whether "item" is a spawner but you don't check if "i" is one too.

In the method isItem() you have to check if "i" is a spawner. "i" is available as "this" in that method's context. This is basic Object Oriented Programming.

Reply:
And by "Show me that code" I actually meant the tests in which a SlimefunItem's id would suddenly be MOB_SPAWNER or ANVIL, which is outright impossible.
Perhaps, you confuse getID() with item.getType()